src/components/loader-spinner/loader-spinner.spec.tsx

```diff
@@ -56,6 +56,20 @@ describe("LoaderSpinner", () => {
   });
 
   describe("when custom props are passed", () => {
+    it("when the 'spinnerLabel' prop is passed a custom string value it overrides the default visible label", () => {
+      render(<LoaderSpinner spinnerLabel="foo" />);
+      const visibleLabelElement = screen.getByTestId("visible-label");
+
+      expect(visibleLabelElement).toHaveTextContent("foo");
```

Comment:
**suggestion(non-blocking)**: what do you think about querying if the label text is visible instead of using test ids? That way, it would mimick how a visible user would see the label:

```suggestion
      const visiblelabel = screen.getByText("foo");

      expect(visibleLabel).toBeVisible();
```

Reply:
I wouldn't be opposed to this necessarily, but the previous approach dictates that we're actually testing the text is contained visually hidden labels DOM element, whereas in theory this is just asserting any text exists in the DOM - potentially anywhere

Reply:
I think this would be intentional here, as a visually-abled user won't care about which DOM element inside of the `LoaderSpinner` the text is placed in, but rather that the loading text is visible to them.

Reply:
I get what you mean mate, but this visible label is still announced to screen-reader users as it's rendered within the implicit aria-live region. The visually alternative hidden label is only rendered in its stead and announced when the `showSpinnerLabel` prop is `true`